### ISSUE TITLE
Actor Reminder fires only once after restart 

### ISSUE THREAD

**Issue by javageek79** [dapr/dapr#2554]
## In what area(s)?

/area runtime
> /area operator
/area placement
> /area docs
> /area test-and-release

## What version of Dapr?

1.0.0-rc.1
> 0.2.x
> Output of git describe --dirty

## Expected Behavior
After a pod/sidecar restart reminders have been restored and are getting fired according to their duration and period settings.

## Actual Behavior

after a sidecar restart, the `receiveReminder()` method is only called once per persisted reminder. Subsequent tries produces error log entries.
````
== DAPR == time="2020-12-09T15:17:36.3693303+01:00" level=debug msg="placement order received: lock" app_id=polling instance=LOCALHOST scope=dapr.runtime.actor.internal.placement type=log ver=1.0.0-rc.1

== DAPR == time="2020-12-09T15:17:36.3713306+01:00" level=debug msg="placement order received: update" app_id=polling instance=LOCALHOST scope=dapr.runtime.actor.internal.placement type=log ver=1.0.0-rc.1

== DAPR == time="2020-12-09T15:17:36.3733317+01:00" level=debug msg="executing reminder poll-reminder2 for actor type DevicePollingActor with id ACTOR_ID" app_id=polling instance=LOCALHOST scope=dapr.runtime.actor type=log ver=1.0.0-rc.1

== DAPR == time="2020-12-09T15:17:36.3753299+01:00" level=debug msg="executing reminder poll-reminder5 for actor type DevicePollingActor with id ACTOR_ID" app_id=polling instance=LOCALHOST scope=dapr.runtime.actor type=log ver=1.0.0-rc.1

== DAPR == time="2020-12-09T15:17:36.3763298+01:00" level=info msg="placement tables updated, version: 41" app_id=polling instance=LOCALHOST scope=dapr.runtime.actor.internal.placement type=log ver=1.0.0-rc.1

== DAPR == time="2020-12-09T15:17:36.3763298+01:00" level=debug msg="placement order received: unlock" app_id=polling instance=LOCALHOST scope=dapr.runtime.actor.internal.placement type=log ver=1.0.0-rc.1

== DAPR == time="2020-12-09T15:17:36.3763298+01:00" level=debug msg="executing reminder poll-reminder5 for actor type DevicePollingActor with id ACTOR_ID" app_id=polling instance=LOCALHOST scope=dapr.runtime.actor type=log ver=1.0.0-rc.1

== DAPR == time="2020-12-09T15:17:36.5912089+01:00" level=error msg="could not find active reminder with key: DevicePollingActor||ACTOR_ID||poll-reminder5" app_id=polling instance=LOCALHOST scope=dapr.runtime.actor type=log ver=1.0.0-rc.1

== DAPR == time="2020-12-09T15:17:36.6012258+01:00" level=error msg="could not find active reminder with key: DevicePollingActor||ACTOR_ID||poll-reminder2" app_id=polling instance=LOCALHOST scope=dapr.runtime.actor type=log ver=1.0.0-rc.1

== DAPR == time="2020-12-09T15:17:36.7932072+01:00" level=error msg="could not find active reminder with key: DevicePollingActor||ACTOR_ID||poll-reminder5" app_id=polling instance=LOCALHOST scope=dapr.runtime.actor type=log ver=1.0.0-rc.1
```` 

## Steps to Reproduce the Problem

connect dapr to either cosmosDB or local redis statestore. Register reminder with a period of 2sec
stop and start sidecar.

## Release Note

RELEASE NOTE:

**Comment by artursouza** [comment]
@javageek79 Thanks for reporting this. I've written an integration test testing the restart of the sidecar and the reminder continues to work. This issue seems to be related to something else. As per the logs pasted above, the issue happens after placement table is updated:
```
== DAPR == time="2020-12-09T15:17:36.3763298+01:00" level=info msg="placement tables updated, version: 41" app_id=polling instance=LOCALHOST scope=dapr.runtime.actor.internal.placement type=log ver=1.0.0-rc.1

== DAPR == time="2020-12-09T15:17:36.3763298+01:00" level=debug msg="placement order received: unlock" app_id=polling instance=LOCALHOST scope=dapr.runtime.actor.internal.placement type=log ver=1.0.0-rc.1
```
It seems that the given actor id got moved to another instance but the reminder kept firing. Can you check if the reminder for the same actorId was fired in another instance?

**Comment by javageek79** [comment]
Hi @artursouza, as I took the logs from my local system, instance was the same after restart. On k8s, behaviour is the same. After restart, reminder is triggered and executed only once. The secon try produces the decribed error.

**Comment by artursouza** [comment]
OK, the issue could not be reproduce out of master branch but was reproduced with the rc1 release. The reason is because this bug has been fixed already in #2049. Closing as duplicate.
